### PR TITLE
move benchmark utils from SolverTools

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

File: src/bmark_utils.jl [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+end
+
+"""
+    statuses, avgs = quick_summary(stats; kwargs...)
+
+Call `count_unique()` and compute a few average measures for each solver in `stats`.
+
+#### Arguments
+
+* `stats::Dict{Symbol,DataFrame}`: benchmark statistics such as returned by `bmark_solvers()`.
```
Comment (abelsiqueira) [inline src/bmark_utils.jl:92]:
```suggestion
* `stats::Dict{Symbol,DataFrame}`: benchmark statistics such as returned by `bmark_solvers`.
```

File: src/bmark_utils.jl [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+#### Keyword arguments
+
+* `cols::Vector{Symbol}`: symbols indicating `DataFrame` columns in solver statistics for which we
+  compute averages. Default: `[:iter, :neval_obj, :neval_grad, :neval_hess, :neval_hprod, :elapsed_time]`.
+
+#### Return value
+
+* `statuses::Dict{Symbol,Dict{Symbol,Int}}`: a dictionary of number of occurrences of each final
+  status for each solver in `stats`. Each value in this dictionary is returned by `count_unique()`
```
Comment (abelsiqueira) [inline src/bmark_utils.jl:102]:
```suggestion
  status for each solver in `stats`. Each value in this dictionary is returned by `count_unique`
```

File: src/run_solver.jl [hunk 3]
```diff
@@ -0,0 +1,86 @@
+export solve_problems
+
+"""
+    solve_problems(solver, problems :: Any; kwargs...)
```
Comment (abelsiqueira) [inline src/run_solver.jl:4]:
```suggestion
    solve_problems(solver, problems; kwargs...)
```

File: src/run_solver.jl [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+#### Return value
+* a `DataFrame` where each row is a problem, minus the skipped ones if `prune` is true.
+"""
+function solve_problems(solver, problems;
+                        solver_logger :: AbstractLogger=NullLogger(),
+                        reset_problem :: Bool = true,
+                        skipif :: Function=x->false,
+                        colstats :: Vector{Symbol} = [:name, :nvar, :ncon, :status, :elapsed_time, :objective, :dual_feas, :primal_feas],
+                        info_hdr_override :: Dict{Symbol,String} = Dict{Symbol,String}(),
+                        prune :: Bool=true, kwargs...)
```
Comment (abelsiqueira) [inline src/run_solver.jl:30]: `reset_problem`, `colstats` and `info_hdr_override` aren't described in the docstring.

**Comment by dpo** [comment]
> I guess the main is whether we want to constrain the benchmark to NLPModels and SolverTools. How well does this work for comparing Krylov methods, for instance?
Although maybe we could incorporate the stats from Krylov into SolverTools.

We haven't been comparing Krylov solvers based on the stats. We've been comparing them based on the metrics accumulated by PkgBenchmark (gc time, allocations, etc.).

This PR isn't making SB less general than it was. But it's true that we should be able to benchmark other solvers than optimization solvers, and thus use other sets of stats.

**Comment by dpo** [comment]
Updated. I think I moved everything.

File: test/test_bmark.jl [hunk 5]
```diff
@@ -0,0 +1,75 @@
+using DataFrames
+using Logging
+using NLPModels
+using SolverTools
+
+# TODO: remove the SolverBenchmark. qualifiers below
+# when the corresponding methods have been removed
+# from SolverTools.
+
```
Comment (abelsiqueira) [inline test/test_bmark.jl:9]:
```suggestion

```

File: Project.toml [hunk 6]
```diff
@@ -25,16 +29,20 @@ BenchmarkTools = "^0.4.2, 0.5"
...
 GitHub = "^5.0.2"
+JLD2 = "0.1.12"
 JSON = "0.20.0, 0.21.0"
 LaTeXTabulars = "^0.1.0"
+NLPModels = "0.13"
 PkgBenchmark = "^0.2.0"
 Plots = "1.0, 1.1"
 PrettyTables = "0.9.0"
-julia = "^1.0.0"
+SolverTools = "0.2"
```
Comment (abelsiqueira) [inline Project.toml:39]:
```suggestion
SolverTools = "0.3"
```
Do we need SolverTools now that most code is moved?

Comment (dpo) [reply Project.toml:39]: We do, for the logging.

File: test/test_bmark.jl (hunk 5 above)
Comment (dpo) [reply test/test_bmark.jl:9]: First, we need to update ST and release.

Comment (abelsiqueira) [reply test/test_bmark.jl:9]: Done

**Pull request merged**